### PR TITLE
Release of SDK for Synse v3

### PR DESCRIPTION
**Pull request by edaniszewski**
This PR:
- merges in all the changes for synse v3 into master

All these changes have all  been reviewed prior to being committed to the v3/staging branch, so this should only require a look-over.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,33 +1,21 @@
...
-build:  ## Build the SDK locally
-	go build -v ./sdk || exit
-
-.PHONY: ci
-ci:  ## Run CI checks locally (build, test, lint)
-	@$(MAKE) build test lint
 
 .PHONY: clean
 clean:  ## Remove temporary files
-	go clean -v || exit
```
[the remark this answers is not in the document]
Comment (edaniszewski) [reply Makefile]: good call. I'll look into this prior to merge.

File: go.mod [hunk 2]
```diff
@@ -1,19 +1,25 @@
 module github.com/vapor-ware/synse-sdk
 
-go 1.13
+go 1.12
```
[the remark this answers is not in the document]
Comment (edaniszewski) [reply go.mod]: whoops. I think this was a mistake while juggling a bunch of merge conflicts on the deps file. I'll update -- thanks for catching this

File: sdk/health/checks.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+package health
+
+// CheckType defines the type of a health Check.
+type CheckType string
+
+// Check is an interface which all health checks should implement.
+type Check interface {
+	GetName() string
```
[the remark this answers is not in the document]
Comment (edaniszewski) [reply sdk/health/checks.go]: yes! I'll add comments in

**Comment by edaniszewski** [comment]
@MatthewHink many many thanks for the review and catching a bunch of things. started the process of updating based on  your notes

**Comment by MatthewHink** [comment]
Last commit LGTM

**Pull request merged**